app/api/qstash/workers/sync-users/route.ts

```diff
@@ -0,0 +1,104 @@
+// This is a public API endpoint that will be invoked by QStash.
+import { verifySignatureAppRouter } from "@upstash/qstash/dist/nextjs";
+import { NextRequest, NextResponse } from "next/server";
+import { loadQstash } from "../../../../../lib/qstash";
+import { UsersSyncPayload } from "../../../../../lib/qstash/types";
+import prisma from "../../../../../lib/prisma";
+import {
+  fetchFarcasterProfiles,
+  UserProfile,
+} from "../../../../../lib/airstack";
+
+async function handler(request: NextRequest) {
+  // Get the payload from the request and extract the offset
+  const body: UsersSyncPayload = await request.json();
+  const { offset } = body;
+
+  // Declaring the batch size
+  const batchSize = 400;
+
+  try {
+    // Read users from the WaitlistedUser Model in the database, grouped by fid with and taken with an offset
+    const usersBatch = await prisma.waitlistedUser.findMany({
```

Comment:
Instead of this query, have a look at https://www.prisma.io/docs/orm/prisma-client/queries/aggregation-grouping-summarizing#group-by